src/trans.rs

```diff
@@ -573,21 +581,7 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
             // If the function is diverging, handle the panic lang item
             // TODO: when it's possible to print characters or interact with the environment, also
             //       handle #[lang = "panic_fmt"] to support panic messages
-            let mut is_fn_panic = false;
-            if self.sig.output == ty::FnOutput::FnDiverging {
-                let attrs = self.tcx.map.attrs(nid);
-                for attr in attrs {
-                    if let MetaItemKind::NameValue(ref node, ref span) = attr.node.value.node {
-                        if node == "lang" {
-                            if let LitKind::Str(ref value, _) = span.node {
-                                if value == "panic" {
-                                    is_fn_panic = true;
-                                }
-                            }
-                        }
-                    }
-                }
-            }
+            let mut is_fn_panic = Some(self.did) == self.tcx.lang_items.panic_fn();
```

Comment:
This shouldn't be `mut`. In fact, it's only used once, in the condition below. If you switch the "then" and "else" bodies, you can use this comparison as the condition.

Similar story with `is_entry_fn` which could be `Some(nid) == self.node_id` below.